File: app/src/main/java/com/akilimo/mobile/Locales.java
package com.akilimo.mobile;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

public class Locales {
    private static final Locale LOCAL_TZ_SWA = new Locale("sw", "TZ");
    private static final Locale LOCAL_KE_SWA = new Locale("sw", "KE");
    private static final Locale LOCAL_NG_ENGLISH = new Locale("en", "NG");
    private static final Locale LOCAL_RW_KINYARWANDA = new Locale("rw", "RW");

    public static final List<Locale> APP_LOCALES = Arrays.asList(
            Locale.ENGLISH,
            Locales.LOCAL_TZ_SWA);

    public static final List<Locale> LOCALE_COUNTRIES = Arrays.asList(Locales.LOCAL_TZ_SWA, LOCAL_NG_ENGLISH, Locales.LOCAL_RW_KINYARWANDA);
}

File: app/src/main/java/com/akilimo/mobile/Locales.kt
package com.akilimo.mobile

import java.util.Locale


object Locales {

    val swahili: Locale = Locale.forLanguageTag("sw-TZ")
    val english: Locale = Locale.forLanguageTag("en-US")
    val kinyarwanda: Locale = Locale.forLanguageTag("rw-RW")

    val supportedLocales: List<Locale> = listOf(
        english,
        swahili,
        kinyarwanda,
    )

    /**
     * Converts any stored language code (old short codes like "en", "sw", "rw" or full
     * BCP-47 tags like "en-US") to the canonical BCP-47 tag used by the app.
     * Falls back to English if the code is unrecognised.
     */
    fun normalize(code: String): String {
        if (code.isBlank()) return english.toLanguageTag()
        return supportedLocales
            .find { it.toLanguageTag().equals(code, ignoreCase = true) || it.language.equals(code, ignoreCase = true) }
            ?.toLanguageTag()
            ?: english.toLanguageTag()
    }
}